From 31e887f05a60a3922733784bedd52b916b6b6746 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 9 Feb 2014 21:44:03 +0000
Subject: [PATCH 1/5] cross runtime

.. we care about TARGET_OS not OS and TARGET_ARCH not ARCH
---
 runtime/compiler-rt/Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/runtime/compiler-rt/Makefile b/runtime/compiler-rt/Makefile
index ccd83a3..51378c0 100644
--- a/runtime/compiler-rt/Makefile
+++ b/runtime/compiler-rt/Makefile
@@ -77,7 +77,7 @@ ifeq ($(shell test -d $(COMPILERRT_SRC_ROOT) && echo OK),OK)
 # build/install runtime libraries for as many targets as clang was configured to
 # support.
 RuntimeDirs :=
-ifeq ($(OS),Darwin)
+ifeq ($(TARGET_OS),Darwin)
 RuntimeDirs += darwin macho_embedded
 RuntimeLibrary.darwin.Configs := \
 	eprintf.a 10.4.a osx.a ios.a cc_kext.a cc_kext_ios5.a \
@@ -109,7 +109,7 @@ endif
 endif
 
 # On Linux, include a library which has all the runtime functions.
-ifeq ($(OS),Linux)
+ifeq ($(TARGET_OS),Linux)
 RuntimeDirs += linux
 RuntimeLibrary.linux.Configs :=
 
@@ -130,11 +130,11 @@ Runtime32BitConfigs = \
 	ubsan-i386.a ubsan_cxx-i386.a
 
 # We currently only try to generate runtime libraries on x86.
-ifeq ($(ARCH),x86)
+ifeq ($(TARGET_ARCH),x86)
 RuntimeLibrary.linux.Configs += $(Runtime32BitConfigs)
 endif
 
-ifeq ($(ARCH),x86_64)
+ifeq ($(TARGET_ARCH),x86_64)
 RuntimeLibrary.linux.Configs += \
 	builtins-x86_64.a profile-x86_64.a san-x86_64.a asan-x86_64.a \
 	asan_cxx-x86_64.a tsan-x86_64.a msan-x86_64.a ubsan-x86_64.a \
@@ -222,7 +222,7 @@ $(ResourceLibDir)/$1/libclang_rt.%.dylib: \
 	$(Echo) Copying runtime library $1/$$* to build dir
 	$(Verb) cp $(PROJ_OBJ_DIR)/clang_$1/$$*/libcompiler_rt.dylib $$@
 	$(Echo) Fixing LC_ID_DYLIB of $$@
-	$(Verb) install_name_tool $$@ -id $$@
+	$(Verb) ${prefix}/bin/$(TARGET_TRIPLE)-install_name_tool $$@ -id $$@
 RuntimeLibrary.$1: \
 		$(RuntimeLibrary.$1.Configs:%=$(ResourceLibDir)/$1/libclang_rt.%)
 .PHONY: RuntimeLibrary.$1
-- 
2.0.0

